### ISSUE TITLE
How to verify all DAX in a model?

### ISSUE THREAD

**Issue by fmms** [otykier/TabularEditor#339]
Hi,

I am deploying with 
```
       $server = "asazure://$(aaRegion).asazure.windows.net/$(aaServerName):rw"
        Write-Host "Deploying to $server and database $(aaDatabase)"
        $p = Start-Process -FilePath "C:\Program Files (x86)\TabularEditor\TabularEditor.exe" -NoNewWindow -PassThru -Wait -ArgumentList "$(Agent.BuildDirectory)/beye/as/Model.asdatabase -VSTS -DEPLOY $server $(aaDatabase) -LOGIN $(asAdminUser) $(asAdminPassword) -OVERWRITE -CONNECTIONS -ROLES"
        Write-Host "finished executing process"
        exit $p.ExitCode
```

This works very nicely. However, the error reporting seems to be strange.

Sometimes there are warning emitted on deployment:
![image](https://user-images.githubusercontent.com/908734/66464448-4c198d80-ea7f-11e9-8869-3892fee7a509.png)

However, these do not concern the current deployment but what was deployed PREVIOUSLY and thus will be overwritten in the very same step.

It would be awesome if we could verify the model we are just deploying.

Thanks

**Comment by otykier** [comment]
Are you saying that the outputted errors no longer exists, once the deployment completes?

That should not be the case. Are you sure that these errors aren't outputted from another step in the build pipeline or a different line in your PowerShell script?

Could you check what happens if you manually execute the deployment from the command line in 2 scenarios:

A) Existing model doesn't have any DAX errors - deployed model has errors
B) Existing model has DAX errors - deployed model doesn't

You can verify if a currently deployed model has errors by connecting to it from within the UI. Tabular Editor will show the error count at the bottom of the screen.

**Comment by fmms** [comment]
Hi ottikier,

yes I am saying exactly that.

For you scenerios:
A)
![image](https://user-images.githubusercontent.com/908734/66682837-55207f80-ec76-11e9-9043-c654d2ab5240.png)

B)
![image](https://user-images.githubusercontent.com/908734/66682776-33bf9380-ec76-11e9-80b0-7909e6759687.png)

And yes Tabular Editor recognizes the error I introduced without any issue:
![image](https://user-images.githubusercontent.com/908734/66684138-a2522080-ec79-11e9-9767-41ba3105e7ef.png)

**Comment by otykier** [comment]
Hang on - your screenshots for A and B looks as I would expect. Does Tabular Editor still show the error when connecting to the model after the deployment in scenario B?

**Comment by fmms** [comment]
I Otykier, sorry for the confusion. It seems the while obfuscating the screenshots. I pasted them in the wrong locations. I have updated the comment and double checked now. In addition I have rephrased your sentences to be even more clear for cases A) and B).

**Comment by fmms** [comment]
Looking at the code I assume we are in the code path at https://github.com/otykier/TabularEditor/blob/e72325c71c606b1ff932bf53bc3a88f9f6b7485e/TOMWrapper/TOMWrapper/TabularDeployer.cs#L109 ? Is that correct?
Logically it seems to be correct, if s.Refresh() is really behaving as expected...

I assume you have tested this in other constellations. So just to give some more background we are using Azure Analysis Services and the model is in compatibility level 1400.

Would it be possible to check a Model.bim file without deployment for any errors? So exposing CheckErrors (https://github.com/otykier/TabularEditor/blob/a1fb16fed4d501520ba88a479e28b1413fd75fc9/TOMWrapper/TOMWrapper/TabularModelHandler.cs#L205 ) from the command line?

**Comment by otykier** [comment]
Checking a model for errors without deployment is not possible. Tabular Editor simply reports the error messages provided by Analysis Services AFTER the deployment happens. For that reason, what you’re describing really shouldn’t happen, as the Model.bim file doesn’t contain any error messages on its own, even if it has invalid DAX syntax. The only way Tabular Editor would know about the errors is after a deployment succeeds...

**Comment by fmms** [comment]
Hi @otykier so any idea what could be the reason? I am presenting next week at SQL Saturday Munich about continuous integration and modern cloud solutions. I would very much like to have a perfectly working tabular editor integration for that.

More background on my context:
- Azure Analysis Services
- Compatibility level 1400
- It is a very big model: 61.500 lines of model.bim, 2000 measures.
- I am using tabular editor 2.8.5
- I am using a MSFT hosted agent and thus installing tabular editor on each run.
- i can reproduce it every time. last time reproduced 1s ago.

**Comment by otykier** [comment]
Fixed in [2.8.6](https://github.com/otykier/TabularEditor/releases/tag/2.8.6).

**Comment by fmms** [comment]
Thanks a lot. I can confirm this to be working now:
![image](https://user-images.githubusercontent.com/908734/66894468-b89d0b00-eff0-11e9-831c-33958ed2e504.png)
